### PR TITLE
Replace `Module["asm"]` with `wasmExports` global

### PR DESCRIPTION
**Pull request by sbc100**
This brings the regular runtime closer `MINIMAL_RUNTIME` which uses the global `asm` to refer to exports object.  As a followup we should consider merging these two symbols.

This also matches the names of the other globals such as `wasmImports` and `wasmMemory`.

### CODE REVIEW

**Comment by sbc100** [comment]
If we think its necessary we could consider continuing to support the old `Module['asm']` syntax property when not in `STRICT` mode?

**Comment by sbc100** [comment]
> If we think its necessary we could consider continuing to support the old `Module['asm']` syntax property when not in `STRICT` mode?

I added support for a `-sEXPORT_RUNTIME_METHODS=wasmImports` so folks can still get access to this from the outside if they really need too.

File: src/preamble.js [hunk 1]
```diff
@@ -979,18 +980,20 @@ function createWasm() {
...
 #else
-    registerTLSInit(Module['asm']['_emscripten_tls_init']);
+    registerTLSInit(wasmExports['_emscripten_tls_init']);
 #endif
 #endif
 
 #if !IMPORTED_MEMORY
-    wasmMemory = Module['asm']['memory'];
+    wasmMemory = wasmExports['memory'];
+    {{{ maybeExport('wasmMemory') }}}
```
Comment (kripken) [inline src/preamble.js:996]: Do we still need

https://github.com/emscripten-core/emscripten/blob/8b5c8687226a27e856f0449db5a92684c4614571/src/modules.js#L384

after this? I think that line can be removed as it doesn't work, as you found.

File: src/parseTools.js [hunk 2]
```diff
@@ -865,6 +865,13 @@ function hasExportedSymbol(sym) {
   return WASM_EXPORTS.has(sym);
 }
 
+function maybeExport(sym) {
```
Comment (kripken) [inline src/parseTools.js:868]: How is this related to the function here with the same name?

https://github.com/emscripten-core/emscripten/blob/3e780ad0f69e97f6883616d108e1c99c23107100/src/modules.js#L347

File: src/preamble.js (hunk 1 above)
Comment (sbc100) [reply src/preamble.js:996]: I tried that, but without adding to the list is modules.js we don't get the `Aborted('wasmExports' was not exported. add it to EXPORTED_RUNTIME_METHODS)` messed when its not exported... and the test currently depends on that.

Comment (kripken) [reply src/preamble.js:996]: I see. Can we perhaps move that logic to a shared place, and have `maybeExport` include it for that assertion? Can be a followup though.

Comment (sbc100) [reply src/preamble.js:996]: Perhaps... `maybeExports` is only needed for these very few items though.. where as the list in `modules.js` is that list of all possible runtime exports.   I'm not sure it makes sense to unify those two things.

File: src/parseTools.js (hunk 2 above)
Comment (sbc100) [reply src/parseTools.js]: I renamed the new function `receivedSymbol` for now, and added a comment.

We could potentially clean this up as a followup by introducing the concept of a delayed export (i.e. and export that cannot be set until after instantiation.    But for now I think this change is big enough.

**Review by kripken — APPROVED** [review]
I don't have a better idea for a name for the new function... if we think of one maybe we can refactor later. Looks great for now.
